### DISCUSSION TITLE
Rename function

### DISCUSSION THREAD

**Discussion by dexman545** [jnr/jnr-ffi#322] [Q&A] [answered]
Is there a way to rename the functions we define in the library interface?
 eg. to give functions with "short" names something more sensible than what they are called in C?
```java
interface Lib {
  void ffghdn(int a);
}
```

**Comment by basshelal** [comment, marked as answer]
Unfortunately not yet, this is something I have thought about and we will try to add in the future.

Something like the below is what I have in mind:
```java
interface Lib {
  @NativeName("ffghdn")
  void myNativeFunction(int a);
}
```

For now, though, use the ugly C native name.
Hope this helps

**Reply by dexman545** [reply]
Thanks, yeah an annotation for this would be nice.
